SConstruct

```diff
@@ -1074,207 +1108,212 @@ env['python_cmd_esc'] = quoted(env['python_cmd'])
-            print(err, err.output)
-        warn_no_python = True
+            sys.exit(1)
+        else:
+            print('WARNING: Problem checking for Python:')
+            print(err)
+            print('Continuing with default parameters.')
     else:
-        (env['python_version'], numpy_version,
-         env['python_usersitepackages']) = info.splitlines()[-3:]
-        numpy_version = LooseVersion(numpy_version)
-        if numpy_version == LooseVersion('0.0.0'):
-            python_message += "NumPy for Python {0} not found.\n".format(env['python_version'])
+        major = env['python_version'][0]
+        py_pkg = 'python{}_package'.format(major)
+        if env[py_pkg] in ['full', 'minimal']:
+            print("ERROR: The python_package points to Python {v} and python{v}_package has been "
+                  "specified. Please change python_package to point to a different version of "
+                  "Python or disable it.".format(v=major))
+            sys.exit(1)
+        elif env[py_pkg] == 'default':
+            py_vars = {'python{}_array_home': '', 'python{}_cmd': 'python{}'.format(major),
+                       'python{}_prefix': '$prefix'}
+
+            env[py_pkg] = env['python_package']
+            for key, value in py_vars.items():
+                if env[key.format(major)] != value:
+                    print("WARNING: The value for {} has been set and will be used instead "
+                          "of {}".format(key.format(major), key.format('')))
+                else:
+                    env[key.format(major)] = env[key.format('')]
+
+    # Make sure everything gets converted to properly versioned variables by deleting
+    # these so they don't get used accidentally
+    del env['python_package']
+    del env['python_array_home']
+    del env['python_cmd']
+    del env['python_prefix']
+    del env['python_version']
+
+for v in [2, 3]:
+    warn_no_python = False
+    python_message = ''
+    env['install_python{}_action'.format(v)] = ''
+    py_pkg = 'python{}_package'.format(v)
+    if env[py_pkg] != 'none':
+        # The directory within the source tree which will contain the Python 2 module
+        env['pythonpath_build{}'.format(v)] = Dir('build/python{}'.format(v)).abspath
+        if 'PYTHONPATH' in env['ENV']:
+            env['pythonpath_build{}'.format(v)] += os.path.pathsep + env['ENV']['PYTHONPATH']
+
+    if env[py_pkg] in ['full', 'default']:
+        # Test to see if we can import the specified array module
+        script = '\n'.join(("from __future__ import print_function",
+                            "import sys",
+                            "print('{v.major}.{v.minor}'.format(v=sys.version_info))",
+                            "try:",
+                            "    import numpy",
+                            "    print(numpy.__version__)",
+                            "except ImportError as err:",
+                            # "    print(err, file=sys.stderr)",
```

Comment:
Why not print the error?

Reply:
The error is reported later, so the printing here is duplicate (and not very clear, because it doesn't say which interface is reporting the error).